cranelift-sizeless-vector.md

```diff
@@ -96,17 +99,19 @@ With the notion of a sizeless stack slot possible, but not a sizeless spill slot
 # Rationale and alternatives
 [rationale-and-alternatives]: #rationale-and-alternatives
 
-The main design decision is to have a vector type that isn't comparable to the existing vector types, which means that no existing paths can accidently try to treat them as such. Although setting the minimum size of the opaque type to 128 bits still allows us to infer the minimum number of lanes for a given lane type. The flexible vector specification also provides specific operations for lane accesses and shuffles so these aren't operations that we have to handle with our existing operations.
+The main change here is the introduction of dynamically created types, using an existing vector type as a base and a scaling factor represented by a global value. Using a global value fits with clif IR in that we have a value which is not expected (allowed?) to change during the execution of the function. The alternative is to add types which have an implicit scaling factor which could make verification more complicated, or impossible.
```

Comment:
Indeed, "not allowed" rather than just "not expected" -- the semantics are as-if loaded once in the prologue.